docs/discord/update.py

```diff
@@ -0,0 +1,170 @@
+                                inCode = not inCode
+                                # make sure it is on a dedicated line
+                                if inCode and not line.startswith('```'):
+                                    line = line.replace('```', '\n```')
+                                if not inCode and len(line) > 3:
+                                    line = line.replace('```', '\n```')
+                            # not inside a multi-line code
+                            if not inCode:
+                                # remove problematic parts
+                                line = line.replace('<i>', '`<i>`')
+                                if line.startswith('#'):
+                                    line = line.replace('#', '\\#')
+                                # protect underscores
+                                undescoreProtected = False
+                                for start, code, end in re.findall(r'([^`]*)`([^`]*)`([^`]*)', line):
+                                    line = line.replace(start, start.replace('_', '\\_'))
+                                    line = line.replace(end, end.replace('_', '\\_'))
+                                    undescoreProtected = True
+                                if not undescoreProtected:
+                                    line = line.replace('_', '\\_')
+                                # make url links
+                                for url in re.findall(r'(?P<url>https?://[^\s]+)', line):
+                                    line = line.replace(url, '[%s](%s)' % (url, url.replace('\\_', '_')))
+                                line += '\n'
+                            # add line to the content
+                            content += line + '\n'
+                        # remove last new line
+                        content = content[:-2]
+                        # replace mention by actual name
+                        for mention in message.mentions:
+                            alternativeMention = mention.mention.replace('<@', '<@!')
+                            content = content.replace(alternativeMention, '`@' + mention.name + '`')
+                            content = content.replace(mention.mention, '`@' + mention.name + '`')
+                        file.write(content)
+                    # add attachments
+                    for attachment in message.attachments:
+                        if (attachment.filename.endswith('.png') or
+                                attachment.filename.endswith('.jpg') or
+                                attachment.filename.endswith('.jpeg') or
+                                attachment.filename.endswith('.gif') or
+                                attachment.filename.endswith('.bmp') or
+                                attachment.filename.endswith('.PNG') or
+                                attachment.filename.endswith('.JPG') or
+                                attachment.filename.endswith('.JPEG')):
```

Comment:
Not all uppercase cases are handled here, like ".GIF", ".BMP".
Why not using something like:
```
name, extension = os.path.splitext(attachment.filename)
if extension.lower() in ['bmp', 'gif', 'jpg', 'jpeg', 'png']:
```
which is shorter and more readable.